packages/hadron-app-registry/src/register-plugin.tsx

```diff
@@ -144,11 +144,90 @@ const useMockOption = <T extends keyof MockOptions>(key: T): MockOptions[T] => {
   return useContext(MockOptionsContext)[key];
 };
 
+function useHadronPluginActivate<T, S extends Record<string, () => unknown>>(
+  config: HadronPluginConfig<T, S>,
+  services: S | undefined,
+  props: T
+) {
+  const registryName = `${config.name}.Plugin`;
+  const isMockedEnvironment = useMockOption('mockedEnvironment');
+  const mockServices = useMockOption('mockServices');
+
+  const globalAppRegistry = useGlobalAppRegistry();
+  const localAppRegistry = useLocalAppRegistry();
+
+  const serviceImpls = Object.fromEntries(
+    Object.keys({
+      ...(isMockedEnvironment ? mockServices : {}),
+      ...services,
+    }).map((key) => {
+      try {
+        return [
+          key,
+          isMockedEnvironment && mockServices?.[key]
+            ? mockServices[key]
+            : services?.[key](),
+        ];
+      } catch (err) {
+        if (
+          err &&
+          typeof err === 'object' &&
+          'message' in err &&
+          typeof err.message === 'string'
+        )
+          err.message += ` [locating service '${key}' for '${registryName}']`;
+        throw err;
+      }
+    })
+  ) as Services<S>;
+
+  const [{ store, actions }] = useState(
+    () =>
+      localAppRegistry.getPlugin(registryName) ??
+      (() => {
+        const plugin = config.activate(
+          props,
+          {
+            globalAppRegistry,
+            localAppRegistry,
+            ...serviceImpls,
+          },
+          createActivateHelpers()
+        );
+        localAppRegistry.registerPlugin(registryName, plugin);
+        return plugin;
+      })()
+  );
+
+  return { store, actions };
+}
+
 export type HadronPluginComponent<
   T,
   S extends Record<string, () => unknown>
 > = React.FunctionComponent<T> & {
   displayName: string;
+
+  /**
+   * Hook that will activate plugin in the current rendering scope without
+   * actually rendering it. Useful to set up plugins that are rendered
+   * conditionally and have to subscribe to event listeners earlier than the
+   * first render in their lifecycle
+   *
+   * @example
+   * const Plugin = registerHadronPlugin(...);
+   *
+   * function Component() {
+   *   Plugin.useActivate();
+   *   const [pluginVisible] = useState(false);
+   *
+   *   // This Plugin component will already have its store set up and listening
+   *   // to the events even before rendering
+   *   return (pluginVisible && <Plugin />)
+   * }
+   */
+  useActivate(props: T): void;
```

Comment:
Should this return `store` or otherwise allow the plugin to export an interface for interacting with it?

Reply:
It does, this was really more of an open question but your reasoning makes sense to me :+1: